dxr/plugins/rust/tests/test_fns/test_fns.py

```diff
@@ -0,0 +1,82 @@
+from dxr.testing import DxrInstanceTestCase
+
+class FunctionTests(DxrInstanceTestCase):
+    def test_simple_function(self):
+        self.found_line_eq('function:foo', "fn <b>foo</b>() {", 3)
+
+    def test_case_insensitive(self):
+        self.found_line_eq('function:FOO', "fn <b>foo</b>() {", 3,
+                           is_case_sensitive=False)
+
+    def test_methods_are_functions_too(self):
+        self.found_line_eq('function:baz', "fn <b>baz</b>(&amp;self) {}", 10)
+        self.found_lines_eq('function:qux',
+                            [('fn <b>qux</b>(&amp;self);', 16),
+                             ('fn <b>qux</b>(&amp;self) {}', 23)])
+        self.found_line_eq('function:norf', 'fn <b>norf</b>(&amp;self) {', 17)
+
+    def test_multiple_with_same_name(self):
+        self.found_lines_eq('function:bar',
+                            [("fn <b>bar</b>() {}", 12),
+                             ("fn <b>bar</b>() {}", 26)])
+
+    def test_fn_ref(self):
+        self.found_line_eq('function-ref:foo', "<b>foo</b>();", 39)
+
+    def test_method_ref(self):
+        self.found_line_eq('function-ref:baz', "b.<b>baz</b>();", 42)
+        self.found_lines_eq('function-ref:qux',
```

Comment:
It looks like most of these are referenced from different places and in different ways, so maybe each of these search results tests a distinct code path. If not, though, don't feel shy about breaking things out into tiny, 10-line SingleFileTestCases. Experience has shown me that many small, independent test fixtures are less brittle and more understandable than a few big ones. I leave it to your judgement whether you're at a good place on the continuum. It sure looks like you're thorough, at least! :-D